File: P6-NascarAngularPage/src/app/componentes/LecturaMarcaDTO.ts
import { LecturaAutoDTO } from "./pilotos/LecturaPilotoDTO";

export interface LecturaMarcaDTO {
    id: number;
    nombre: string;
    foto: string;
    listaAutos: LecturaAutoDTO[];
}

export interface LecturaMarcaNombreDTO {
    id: number;
    nombre: string;
    foto: string;
}

export interface CrearMarcaDTO{
    nombre: string;
    foto: File | null | undefined;
}

File: P5-NascarSeasonWebAPI/NascarPage/NascarPage/DTOs/LecturaMarcaDTO.cs
using System.ComponentModel.DataAnnotations;

namespace NascarPage.DTOs
{
    public class LecturaMarcaDTO
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = null!;
        public string Foto { get; set; } = null!;
        public List<LecturaAutoDTO> ListaAutos { get; set; } = null!;
    }

    public class LecturaMarcaNombreDTO
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = null!;
        public string Foto { get; set; } = null!;
    }
}
